### PR TITLE
SUP-5504 Add private_ip variable to terraform-aws-tamr-vm

### PR DESCRIPTION
**Pull request by along528**
Allow user to pass list of `private_ips` to the module to be passed to the aws_network_interface attached to the ec2 instance

### CODE REVIEW

File: modules/aws-ec2-instance/main.tf [hunk 1]
```diff
@@ -24,6 +24,7 @@ resource "aws_instance" "tamr-instance" {
 
 resource "aws_network_interface" "tamr-instance-network" {
   subnet_id       = var.subnet_id
+  private_ips = var.private_ips
```
Comment (souza-dan) [inline modules/aws-ec2-instance/main.tf:27]: Formatting is off. Run terraform fmt to correct. It has a recursive flag so you can run it at the root of the repo to ensure everything is formatted correctly.

File: variables.tf [hunk 2]
```diff
@@ -129,3 +129,9 @@ variable "emr_abac_valid_tags" {
   description = "A map of valid tags for maintaining EMR resources when using ABAC IAM Policies with Tag Conditions. Make sure your tamr-config.yml file specifies tags key values. Refer to tamr-config module examples for more info."
   default     = {}
 }
+
+variable "private_ips" {
+  type        = list(string)
+  default     = []
```
Comment (souza-dan) [inline variables.tf:135]: Assuming setting this to an empty list is equivalent to the previous behavior. Has that been tested?

**Review by souza-dan — CHANGES_REQUESTED** [review]
Remember to update the minor version and changelog, but otherwise looks good.

File: variables.tf (hunk 2 above)
Comment (along528) [reply variables.tf:135]: Ran a couple more tests and indeed it looks like there is an issue with passing this default to the network interface. While relying on the default did result in it choosing an IP on it's own, later running some changes resulted in the following error
```
Error: Failure to unassign Private IPs: InvalidParameterValue: Value (10.1.0.208) for parameter privateIpAddress is invalid. The primary IP address of an interface cannot be unassigned.
        status code: 400, request id: c5fe7105-51fb-4a8c-9f2f-7dfddb906b1e

  on .terraform/modules/tamr-vm/modules/aws-ec2-instance/main.tf line 25, in resource "aws_network_interface" "tamr-instance-network":
  25: resource "aws_network_interface" "tamr-instance-network" {
```

The error goes away if then specify explicitly the IP that was chosen during the initial deployment.  So it seems need to treat the default another way when passing to the network interface. Is there a preferred way to do this? Perhaps using [nullable arguments](https://www.terraform.io/docs/language/expressions/types.html#null)?

Comment (souza-dan) [reply variables.tf:135]: yeah, setting `null` might be equivalent to what was happening prior to this change

Comment (along528) [reply variables.tf]: Updated defaults to `null` and confirmed works as previously

**Review by souza-dan — APPROVED** [review]
LGTM

**Review by daniosim — APPROVED** [review]

**Pull request merged**